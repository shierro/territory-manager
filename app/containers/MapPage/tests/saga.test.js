import { put, takeLatest } from 'redux-saga/effects';
import {
  SET_INITIAL_LOCATION,
  GET_INITIAL_LOCATION,
  SET_PAGE_ERROR,
  SET_LOADING,
  ADD_PERSON_START,
} from '../constants';

// import { setInitialLocation } from './actions';

import watchMapsPageSagas, { setInitLocation, getLocation } from '../saga';

describe('MapPage Sagas', () => {
  describe('setInitialLocation Saga', () => {
    let generator;
    beforeEach(() => {
      generator = setInitLocation();
      const callDescriptor = generator.next(getLocation).value;
      expect(callDescriptor).toMatchSnapshot();
    });

    it('should dispatch the setTickets action if it requests the data successfully', () => {
      const response = {
        latitude: 5,
        longitude: 5,
      };
      const putDescriptor = generator.next(response).value;
      expect(putDescriptor).toEqual(
        put({
          type: SET_INITIAL_LOCATION,
          coords: [response.latitude, response.longitude],
        }),
      );
    });

    it('should set error action if there is something wrong', () => {
      const error = 'Some error';
      const putDescriptor = generator.throw(error).value;
      expect(putDescriptor).toEqual(put({ type: SET_PAGE_ERROR, error }));
      expect(generator.next().value).toEqual(
        put({ type: SET_LOADING, value: false }),
      );
    });
  });

  describe('watchMapsPageSagas Saga', () => {
    const sagas = watchMapsPageSagas();
    it('should start task to watch ALL actions', () => {
      const takeLatestDescriptor = sagas.next().value;
      expect(takeLatestDescriptor).toEqual(
        takeLatest(GET_INITIAL_LOCATION, setInitLocation),
      );
      const takeLatestDescriptorSecond = sagas.next().value;
      expect(takeLatestDescriptorSecond).toEqual(
        takeLatest(ADD_PERSON_START, setInitLocation),
      );
    });
  });
});
