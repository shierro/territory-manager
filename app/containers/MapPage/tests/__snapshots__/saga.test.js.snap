// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MapPage Sagas setInitialLocation Saga should dispatch the setTickets action if it requests the data successfully 1`] = `
Object {
  "@@redux-saga/IO": true,
  "CALL": Object {
    "args": Array [],
    "context": null,
    "fn": [Function],
  },
}
`;

exports[`MapPage Sagas setInitialLocation Saga should set error action if there is something wrong 1`] = `
Object {
  "@@redux-saga/IO": true,
  "CALL": Object {
    "args": Array [],
    "context": null,
    "fn": [Function],
  },
}
`;
